### ISSUE TITLE
False OK on raid controller

### ISSUE THREAD

**Issue by petski** [ercpe/smart-check#5]
Given a MR9340-8i raid controller, with a ST4000NM0023 disk in slot 15:

```
[root@redacted ~]# /usr/bin/smart-check --debug --verbose /dev/sda; echo exitcode: $?
DEBUG    Executing smartctl command:  /usr/sbin/smartctl  -n standby -a /dev/sda
DEBUG    Failed generic attributes: {}
DEBUG    Device does not exist in database
MR9340-8i: S.M.A.R.T. data OK
exitcode: 0
```

```
[root@redacted ~]# /usr/bin/smart-check --debug --verbose /dev/sda --interface 'megaraid,15'; echo exitcode: $?
DEBUG    Executing smartctl command:  /usr/sbin/smartctl -d megaraid,15 -n standby -a /dev/sda
DEBUG    Failed generic attributes: {}
DEBUG    Device does not exist in database
ST4000NM0023: S.M.A.R.T. data OK
exitcode: 0
```

I'd expect the first executed command to fail, instead of succeed.

Output of the executed command:

```
[root@redacted ~]# /usr/sbin/smartctl  -n standby -a /dev/sda; echo exitcode: $?
smartctl 7.0 2018-12-30 r4883 [x86_64-linux-3.10.0-1127.10.1.el7.x86_64] (local build)
Copyright (C) 2002-18, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF INFORMATION SECTION ===
Vendor:               AVAGO
Product:              MR9340-8i
Revision:             4.62
Compliance:           SPC-3
User Capacity:        4,000,225,165,312 bytes [4.00 TB]
Logical block size:   512 bytes
Logical Unit id:      0x600605b00bf860b026c6a4e8041f90d5
Serial number:        00d5901f04e8a4c626b060f80bb00506
Device type:          disk
Local Time is:        Thu Aug 13 10:10:30 2020 UTC
SMART support is:     Unavailable - device lacks SMART capability.

=== START OF READ SMART DATA SECTION ===
Current Drive Temperature:     0 C
Drive Trip Temperature:        0 C

Error Counter logging not supported

Device does not support Self Test logging
exitcode: 4
```

**Comment by ercpe** [comment]
Thank you very much for the report.

Indeed, `smart-check` should not report OK for devices without SMART support. Previously, `smart-check` relied entirely on the other sections in the `smartctl` output do detect if something is wrong with the device. I've updated the check to test for the SMART support in the information section.

Could you please test it with the `0.5` tag (or the current `master`) and verify that it's working for you as expected?

`smart-check` should exit with code 2 (CRITICAL in Icinga) with the message: ```MR9340-8i: Device is missing S.M.A.R.T. support```

**Comment by petski** [comment]
Works like a charm. Thank you for the fix! (p.s. that was quick! again!)

Two minor remarks:
- `smart-check --version` returns `smart-check 0.4.2` (probably from `__init__.py`) instead of `smart-check 0.5.0` (or something simular)
- A TODO in the code mentions picking up the exit code from `smartctl` (that's why I included it in my outputs). This may indeed even be better than parsing the output. If you need any help with that, let me know. I have tons of hardware to test with.

**Comment by ercpe** [comment]
> smart-check --version returns smart-check 0.4.2 (probably from __init__.py) instead of smart-check 0.5.0 (or something simular)

Whoopsie. That's why you shouldn't release software without being properly caffeinated. I've updated the version string and released a 0.5.1 to correct that issue. No other changes were made.

> A TODO in the code mentions picking up the exit code from smartctl (that's why I included it in my outputs). This may indeed even be better than parsing the output. If you need any help with that, let me know. I have tons of hardware to test with.

IIRC I looked into the exit code when I started the project but I really can't remember why I abandoned the idea at the time. Probably because the exit codes only give us a top-level view and I wanted to include more information in the output.
I split up this topic into https://github.com/ercpe/smart-check/issues/6 to track it. Since I no longer manage servers as my day-to-day job and moved my private stuff elsewhere, I don't really use smart-check anymore so I don't think I will implement this any time soon.
Nevertheless, feel free to open issues if you stumble upon any other bugs - i will do my best to fix them in a timely manner.
